### PR TITLE
Feature/vote 522/json files drupal api

### PR DESCRIPTION
**Pull request by SamiraMSadat**
Jira Ticket: https://cm-jira.usa.gov/browse/VOTE-522

This ticket is the first pass at integrating the json generated by Drupal into the NVRF tool. There will be some missing data that still needs to be defined, so I am leaving in registration.json and navigation.json for now.

### CODE REVIEW

File: src/components/RegType/StepsList.jsx [hunk 1]
```diff
@@ -29,14 +32,13 @@ function StepsList(props) {
...
-    
+
 
     return (
         <>
-
-            <h2>{content.process_heading}</h2>
-            <p>{content.process_parag}</p>
-
+            <div dangerouslySetInnerHTML= {{__html: contentHeading}}/> {/* TODO: Need to style each paragraph */}
```
Comment (rayestrada) [inline src/components/RegType/StepsList.jsx:39]: We don't need html for the heading field. We should have an h2 here. Maybe this needs to be fixed on the CMS side.

File: src/components/PathSelection.jsx [hunk 2]
```diff
@@ -2,38 +2,46 @@ import { Button, CardGroup, Card, CardHeader, CardBody, CardFooter, Icon } from
...
-    const [content, setContent] = useState()
+    const [content, setContent] = useState('')
+    const [cards, setCards] = useState('')
+    const [navContent, setNavContent] = useState('')
+
     useEffect(() => {
-        fetchData("path-selection.json", setContent);
+        fetchData("cards.json", setCards);
+        fetchData("pages.json", setContent);
+        fetchData("navigation.json", setNavContent);
```
Comment (rayestrada) [inline src/components/PathSelection.jsx:15]: pull these from props

File: src/components/FormSections/Delivery.jsx [hunk 3]
```diff
@@ -3,11 +3,12 @@ import React, { useState, useEffect } from "react";
 import {fetchData} from '../HelperFunctions/JsonHelper.jsx';
 import GenerateFilledPDF from '../GenerateFilledPDF';
 import reactStringReplace from 'react-string-replace';
+import DOMPurify from 'dompurify';
 
 function Delivery(props) {
     const [content, setContent] = useState();
     useEffect(() => {
-        fetchData("delivery.json", setContent);
+        fetchData("pages.json", setContent);
```
Comment (rayestrada) [inline src/components/FormSections/Delivery.jsx:11]: pull in from props.

File: src/components/FormSections/Confirmation.jsx [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import { Alert, Button, Checkbox } from '@trussworks/react-uswds';
 import React, { useState, useEffect } from "react";
 import {fetchData} from '../HelperFunctions/JsonHelper.jsx';
+import DOMPurify from 'dompurify';
 
 function Confirmation(props){
     const [content, setContent] = useState();
     useEffect(() => {
-        fetchData("confirmation.json", setContent);
+        fetchData("pages.json", setContent);
```
Comment (rayestrada) [inline src/components/FormSections/Confirmation.jsx:9]: pull in from props

File: src/components/Eligibility.jsx [hunk 5]
```diff
@@ -3,46 +3,45 @@ import { fetchData } from './HelperFunctions/JsonHelper.jsx';
...
-    const [content, setContent] = useState()
+    const stateContent = props.stateData;
+    const [data, setData] = useState('')
+    const [fields, setFields] = useState('')
+    const [navContent, setNavContent] = useState('')
     useEffect(() => {
-        fetchData("eligibility.json", setContent);
+        fetchData("pages.json", setData);
+        fetchData("fields.json", setFields);
+        fetchData("navigation.json", setNavContent);
```
Comment (rayestrada) [inline src/components/Eligibility.jsx:16]: pull in props where appropriate

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: src/components/RegType/StepsList.jsx (hunk 1 above)
Comment (SamiraMSadat) [reply src/components/RegType/StepsList.jsx:39]: Yes this should be fixed on the CMS side. I'll change the json we have committed for now.

**Review by rayestrada — APPROVED** [review]

**Pull request merged**